src/query/storage/m3/dynamic_cluster_test.go

```diff
@@ -0,0 +1,416 @@
+				SetAggregationOptions(defaultTestNs2AggregationOpts)
+)
+
+func TestDynamicClustersInitialization(t *testing.T) {
+	ctrl := xtest.NewController(t)
+	defer ctrl.Finish()
+
+	mockSession := client.NewMockSession(ctrl)
+
+	mapCh := make(nsMapCh, 10)
+	mapCh <- testNamespaceMap(t, []mapParams{
+		{nsID: defaultTestNs1ID, nsOpts: defaultTestNs1Opts},
+		{nsID: defaultTestNs2ID, nsOpts: defaultTestNs2Opts},
+	})
+	mockInitializer := newMockNsInitializer(t, ctrl, mapCh)
+
+	cfg := DynamicClusterNamespaceConfiguration{
+		session:       mockSession,
+		nsInitializer: mockInitializer,
+	}
+
+	opts := NewDynamicClusterOptions().
+		SetDynamicClusterNamespaceConfiguration([]DynamicClusterNamespaceConfiguration{cfg}).
+		SetInstrumentOptions(instrument.NewOptions())
+
+	clusters, err := newDynamicClusters(opts)
+	require.NoError(t, err)
+
+	defer cleanup(clusters)
+
+	requireClusterNamespace(t, clusters, defaultTestNs2ID, ClusterNamespaceOptions{
+		attributes: storagemetadata.Attributes{
+			MetricsType: storagemetadata.AggregatedMetricsType,
+			Retention:   48 * time.Hour,
+			Resolution:  1 * time.Minute,
+		},
+		downsample: &ClusterNamespaceDownsampleOptions{All: true},
+	})
+
+	requireClusterNamespace(t, clusters, defaultTestNs1ID, ClusterNamespaceOptions{
+		attributes: storagemetadata.Attributes{
+			MetricsType: storagemetadata.UnaggregatedMetricsType,
+			Retention:   48 * time.Hour,
+		}})
+
+	requireClusterNamespaceIDs(t, clusters, []ident.ID{defaultTestNs1ID, defaultTestNs2ID})
+}
+
+func TestDynamicClustersWithUpdates(t *testing.T) {
+	ctrl := xtest.NewController(t)
+	defer ctrl.Finish()
+
+	mockSession := client.NewMockSession(ctrl)
+
+	mapCh := make(nsMapCh, 10)
+	nsMap := testNamespaceMap(t, []mapParams{
+		{nsID: defaultTestNs1ID, nsOpts: defaultTestNs1Opts},
+		{nsID: defaultTestNs2ID, nsOpts: defaultTestNs2Opts},
+	})
+	mapCh <- nsMap
+	mockInitializer := newMockNsInitializer(t, ctrl, mapCh)
+
+	cfg := DynamicClusterNamespaceConfiguration{
+		session:       mockSession,
+		nsInitializer: mockInitializer,
+	}
+
+	opts := NewDynamicClusterOptions().
+		SetDynamicClusterNamespaceConfiguration([]DynamicClusterNamespaceConfiguration{cfg}).
+		SetInstrumentOptions(instrument.NewOptions())
+
+	clusters, err := newDynamicClusters(opts)
+	require.NoError(t, err)
+
+	defer cleanup(clusters)
+
+	// Update resolution of aggregated namespace
+	newOpts := defaultTestNs2Opts.
+		SetAggregationOptions(defaultTestNs2AggregationOpts.
+			SetAggregations([]namespace.Aggregation{namespace.NewAggregatedAggregation(
+				namespace.AggregatedAttributes{
+					Resolution:        2 * time.Minute,
+					DownsampleOptions: namespace.NewDownsampleOptions(true),
+				}),
+			}))
+	nsMap = testNamespaceMap(t, []mapParams{
+		{nsID: defaultTestNs1ID, nsOpts: defaultTestNs1Opts},
+		{nsID: defaultTestNs2ID, nsOpts: newOpts},
+	})
+
+	// Send update to trigger watch
+	mapCh <- nsMap
+
+done:
```

Comment:
Can we do this without labels?

Reply:
Ahh, we can just return here, so the label + break is actually unnecessary.